dan_layer/wallet/storage_sqlite/src/reader.rs

```diff
@@ -257,7 +257,15 @@ impl WalletStoreReader for ReadTransaction<'_> {
             query = query.filter(substates::template_address.eq(template_address.to_string()));
         }
         if let Some(substate_type) = by_type {
-            query = query.filter(substates::address.like(format!("{}_%", substate_type.as_prefix_str())));
+            match substate_type {
+                SubstateType::NonFungible => {
```

Comment:
Possible sql injection here, but I think substate_type can't be a generic string, so looks ok

Reply:
Yeah cant be too careful, as_prefix_str returns a static str so it cant be anything other than a hardcoded string, so should be ok - NFT addresses need improving